### PR TITLE
Add tooling updates to NEWS.md

### PR DESCRIPTION
**Pull request by ViralBShah**
Mention debugger, Juno+Debugger integration, C++ FFI and multi-threading. I thought these are all big enough to be mentioned right upfront as the major highlights. Perhaps we should also have a few other major highlights such as jb/functions and broadcasting?

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+     also includes support for plotting and debugging.
+
+   * Experimental support for
+     [multi-threading](http://docs.julialang.org/en/latest/manual/parallel-computing/#multi-threading-experimental). This
+     makes it easy to parallelize loops with independent iterations in
+     a common address space with the `@threads` macro.
+
+   * A convenient C++ FFI in
+     [Cxx.jl](https://github.com/Keno/Cxx.jl). Using it requires
+     building Julia with some additional build flags that will build clang along with llvm.
```
Comment (tkelman) [inline NEWS.md]: I don't think this is the case any more. Still a bit in flux.

**Comment by tkelman** [comment]
Startup time on Windows will be very noticeably better, and the upgraded LLVM version is worth noting.

File: NEWS.md (hunk 1 above)
Comment (ViralBShah) [inline NEWS.md]: The Cxx.jl page still says so. @keno - suggestions?

**Comment by ViralBShah** [comment]
Is there an issue or PR for faster startup on windows, or something I can refer to? Not necessary, but good to link if available.

File: NEWS.md (hunk 1 above)
Comment (tkelman) [inline NEWS.md]: Readme last modified a month ago, but see https://github.com/Keno/Cxx.jl/commits/master and the deps/build.jl script

**Comment by tkelman** [comment]
https://github.com/JuliaLang/julia/issues/16953, though a lot of the discussion happened in a few older issues that were mostly about backtraces

File: NEWS.md (hunk 1 above)
Comment (Keno) [inline NEWS.md]: @tkelman is correct, even though there is still a small bug right now that makes it necessary. Once that's resolved `Pkg.add` will just work though.

**Comment by tkelman** [comment]
Many, many new deprecations are also not mentioned in https://github.com/JuliaLang/julia/blob/master/NEWS.md#deprecated-or-removed

**Comment by ViralBShah** [comment]
I guess the deprecations list should either be complete, or removed altogether. Personally, I don't find the list of all deprecations in NEWS.md useful.

**Comment by tkelman** [comment]
The list in NEWS can have more elaboration on subtleties and backwards-compatibility concerns for porting code, since naively following the warning has been frequently seen to be the wrong way to update packages if they don't want to drop support for old versions of Julia yet.

**Comment by ViralBShah** [comment]
The current one looks not much better than the deprecation warnings. If you can list out the different kinds of subtleties and backwards compatibility, everyone can then put things in the right place.

My guess is that many people will just move on and support 0.5 as the min version within a couple of months of its release, unless we have very clear instructions.

**Comment by timholy** [comment]
I like the idea of highlights, but I wonder if it's a concern that 3/4 of them are about packages rather than improvements in core julia. Of course core changes enable that functionality, but a casual reader won't know that and might think that julia proper has hardly changed since 0.4. ("Julia development slowing down, they rely on advertising community packages just to have enough NEWS!")

I also wonder a bit about picking and choosing among the package ecosystem (should Plots and GLVisualize be in that list? won't they be more widely used than Cxx?), but of course the ones you picked out are indeed tremendously exciting highlights.

**Comment by ViralBShah** [comment]
I don't think the highlights should be restricted to the things I listed, and there are probably other things worth calling out that are of general interest. I am not sure about GLVisualize being of direct wide interest (even though it is fantastic), but Plots.jl certainly deserves a mention. Even the language and compiler improvements could perhaps use more description. 

I think the fact that many of these are packages is just the way they are organized, but the debugger and Cxx FFI could equally be in the julia repo.

**Comment by JeffBezanson** [comment]
Let's just mention multithreading under new language features, and then we can label this new section something like "tooling improvements", since that's basically what it's about.

-1 to a "highlights" section. I don't want to go through the file and debate/decide which items count as highlights.

**Comment by StefanKarpinski** [comment]
I think that we should have a blog post that goes along with the release that highlights the most significant changes with cool examples of what you can do with them. I can work on something.

**Comment by ViralBShah** [comment]
Yes, we can have highlights as a blog post, and change this to tools in the NEWS.

**Comment by ViralBShah** [comment]
I have renamed the section to tools improvements. I am perhaps missing this, but the `jb/functions` seems to be missing a mention. If it is, the performance impact is not clearly explained.

Do we need a performance section?

File: NEWS.md [hunk 2]
```diff
@@ -35,18 +52,35 @@ New language features
...
+     [multi-threading](http://docs.julialang.org/en/latest/manual/parallel-computing/#multi-threading-experimental). Loops
+     with independent iterations can be easily parallelized with the
+     `Threads.@threads` macro.
+
+  * Support for arrays with indexing starting at values different from
+    1. The array types are expected to be defined in packages, but now
+    Julia provides an API for writing generic algorithms for arbitrary
+    indexing schemes ([#16260]).
+
+New architectures
```
Comment (tkelman) [inline NEWS.md]: you're duplicating this

File: NEWS.md [hunk 3]
```diff
@@ -110,43 +144,6 @@ New architectures
   * [Power](https://github.com/JuliaLang/julia/issues?utf8=%E2%9C%93&q=label%3Apower):
     [#16455], [#16404]
 
-Breaking changes
```
Comment (tkelman) [inline NEWS.md]: this should be earlier, not later

Comment (ViralBShah) [inline NEWS.md]: I would have preferred all the breaking things and deprecations to be together. The flow that makes sense to me is - new tools stuff, new language things, new compiler things, library improvements, breaking changes and deprecations.

I can see that perhaps breaking changes and language changes need to be clubbed together - but as I look deeper, there seem to be things in language changes that ought to be in breaking (like case-sensitive using and import).

How can we simplify the structure and the communication in this file?

File: NEWS.md (hunk 2 above)
Comment (ViralBShah) [inline NEWS.md]: What is duplicated?

Comment (tkelman) [inline NEWS.md]: you have this section twice now

File: NEWS.md (hunk 3 above)
Comment (tkelman) [inline NEWS.md]: I think breaking changes are more important than compiler and library improvements for people who will be reading this file. deprecations are important to mention too (and way incomplete here) but can maybe be a bit later.

**Comment by tkelman** [comment]
> Do we need a performance section?

A short one would be a good idea. Explaining the consequences of jb/functions making anonymous functions now fast, better startup time on Windows, and better SIMD from newer LLVM (we're paying for it in compile time, but maybe that's a known bug rather than something to advertise...)?

**Comment by ViralBShah** [comment]
I am unlikely to be able to update this for the next couple of days as I will be flying. If someone wants to clean this up and move the breaking changes back to the original place, I'll be grateful. Otherwise, I'll do it once I am back home.

**Comment by StefanKarpinski** [comment]
Having language changes first seems like the only sane thing to me – you go from most central to least. Starting at tooling seems like going from the outside in, Viral. Any intuitive reason for that?

**Comment by ViralBShah** [comment]
My thinking was having things that more users are likely to find value in as they read this. Perhaps as you said, maybe the blog post is the right place to do this. Please feel free to move things around as necessary.

Can someone check if @JeffBezanson's jb/functions stuff is mentioned and appropriately highlighted. Would be nice to have it in a performance section.

**Comment by JeffBezanson** [comment]
Updated.

**Pull request merged**